dot/state/test_helpers.go

```diff
@@ -0,0 +1,90 @@
+package state
+
+import (
+	"math/big"
+	"math/rand"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/trie"
+
+	"github.com/stretchr/testify/require"
+)
+
+// branch tree randomly
+type testBranch struct {
+	hash  common.Hash
+	depth int
+}
+
+// AddBlocksToState adds blocks to a BlockState up to depth, with random branches
+func AddBlocksToState(t *testing.T, blockState *BlockState, depth int) ([]*types.Header, []*types.Header) {
+	previousHash := blockState.BestBlockHash()
+
+	branches := []testBranch{}
+	r := *rand.New(rand.NewSource(rand.Int63()))
+
+	arrivalTime := uint64(1)
+	currentChain := []*types.Header{}
+	branchChains := []*types.Header{}
+
+	// create base tree
+	for i := 1; i <= depth; i++ {
+		block := &types.Block{
+			Header: &types.Header{
+				ParentHash: previousHash,
+				Number:     big.NewInt(int64(i)),
+				StateRoot:  trie.EmptyHash,
+			},
+			Body: &types.Body{},
+		}
+
+		currentChain = append(currentChain, block.Header)
+
+		hash := block.Header.Hash()
+		err := blockState.AddBlockWithArrivalTime(block, arrivalTime)
+		require.Nil(t, err)
+
+		previousHash = hash
+
+		isBranch := r.Intn(2)
```

Comment:
Do we know the seed will produce 50/50 results? Do we need to set the seed using `rand.Seed()`? Probably good as is just want to double check because I have not used this myself before.

Reply:
the seed should ideally produce 50/50 results, it seems to work when you run the test. it's not super important if it's slightly off since it's just to create random tree branching